pkg/errs-dictionary/errdict.go

```diff
@@ -0,0 +1,120 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errdict
+
+import (
+	"strings"
+
+	"istio.io/pkg/structured"
+)
+
+const (
+	LikelyCauseFirstPrefix = "The likely cause is "
+	LikeyCauseSecondPrefix = "Another possible cause could be "
+)
+
+func fixFormat(s string) string {
+	s = strings.TrimSpace(s)
+	return strings.TrimSuffix(s, ".")
+}
+
+func formatCauses(causes ...string) string {
+	if len(causes) == 0 {
+		return ""
+	}
+	out := LikelyCauseFirstPrefix + fixFormat(causes[0]) + "."
+	for _, c := range causes[1:] {
+		out += LikeyCauseSecondPrefix + fixFormat(c) + "."
+	}
+	return out
+}
+
+// General messages
+const (
+	// Boilerplate messages applicable all over the code base.
+
+	// Action
+	ActionIfErrPersistsContactSupport          = "If this error persists, " + ActionContactSupport
+	ActionIfErrSureCorrectConfigContactSupport = "If you are sure your configuration is correct, " + ActionContactSupport
+	ActionContactSupport                       = "contact support if using managed Istio. Otherwise check if the issue is already fixed in " +
+		"another version of Istio at https://github.com/istio/istio/issues, raise an issue, or ask for help in " +
+		"discuss.istio.io."
+
+	// LikelyCause
+	LikelyCauseAPIServer     = "a problem with the k8s API server"
+	LikelyCauseConfiguration = "an incorrect or badly formatted configuration"
+	LikelyCauseSoftwareBug   = "an issue with the Istio code"
+	LikelyCauseTransience    = "If the error occurred immediately after installation, it is likely permanent."
+
+	// Specific errors, also applicable all over the code base.
+
+	// FailedToGetObjectFromAPIServer
+	ImpactFailedToGetObjectFromAPIServer = "If the error is transient, the impact is low. If permanent, updates for " +
```

Comment:
how can we arbitrarily declare that failed to get from api server == "low impact" or "out of sync control plane"? It could mean very very different things in operator vs istiod, or even different code paths. My wider scoped comment is I am not sure it makes sense to have a single global errdict, rather than localized ones (ie operator errors)

Reply:
sure, fair enough. i think there is probably some common boiler plate but i agree it's going to vary wildly and perhaps it's not worthwhile to define any common strings but make everything per feature. 
the reason i'd argue to keep things in a single file is for consistency of the text across all features. writers and reviewers will have immediate access to all the other error definitions which will make it easier to keep the entries consistent. this is probably more important in the beginning when there aren't a lot of examples, so i propose keeping  a single file to start with, with the different features kept apart, and moving into separate files as the volume of text reaches a point where things are getting messy.

Reply:
moved things around so only the most general boilerplate is left in the common section.